db/sql/sql.go

```diff
@@ -148,16 +159,30 @@ func (s *Storage) SetRoot(ctx context.Context, hash *merkletree.Hash) error {
 // Iterate implements the method Iterate of the interface db.Storage
 func (s *Storage) Iterate(ctx context.Context,
 	f func([]byte, *merkletree.Node) (bool, error)) error {
-	items := []NodeItem{}
-
-	err := s.db.SelectContext(ctx, &items,
-		"SELECT * FROM mt_nodes WHERE mt_id = $1", s.mtId)
+	rows, err := s.db.Query(ctx, `SELECT mt_id, key, type, child_l, child_r, entry, created_at, deleted_at
+		FROM mt_nodes WHERE mt_id = $1`, s.mtId)
 	if err != nil {
 		return err
 	}
-	for _, v := range items {
-		k := v.Key[:]
-		n, err := v.Node()
+	defer rows.Close()
```

Comment:
@olomix could you explain this magic? Should I close `rows` all time? Because, all code that I saw, had not used this. Without this line, I was failing unit tests.